### PR TITLE
bpo-30605: Fix compiling binary regexs with BytesWarnings enabled.

### PR DESCRIPTION
**Pull request by rowillia**
Running our unit tests with `-bb` enabled triggered this failure.

### CODE REVIEW

**Comment by jmphilli** [comment]
+1

File: Lib/test/test_re.py [hunk 1]
```diff
@@ -1368,7 +1368,7 @@ def test_inline_flags(self):
             self.assertTrue(re.match(p, lower_char))
         self.assertEqual(
             str(warns.warnings[0].message),
-            'Flags not at the start of the expression %s' % p
+            "Flags not at the start of the expression '%s'" % p
```
Comment (serhiy-storchaka) [inline Lib/test/test_re.py]: Just use `%r`.

And add a test for bytes pattern `b'A(?i)'`.

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]
Add a test for bytes pattern, a Misc/NEWS entry, and your name in Misc/ACKS.

File: Lib/test/test_re.py (hunk 1 above)
Comment (rowillia) [reply Lib/test/test_re.py]: Done

**Comment by rowillia** [comment]
Thanks for your feedback @serhiy-storchaka !  Addressed your comments.

File: Misc/ACKS [hunk 2]
```diff
@@ -1685,6 +1685,7 @@ Jakub Wilk
 Gerald S. Williams
 Jason Williams
 John Williams
+Roy Willams
```
Comment (serhiy-storchaka) [inline Misc/ACKS]: Typo.

File: Misc/NEWS [hunk 3]
```diff
@@ -646,6 +646,9 @@ Library
 - bpo-29851: importlib.reload() now raises ModuleNotFoundError if the
   module lacks a spec.
 
+- bpo-30605: re.compile() no longer raises a BytesWarning when compiling a
+  bytes instance.
```
Comment (serhiy-storchaka) [inline Misc/NEWS]: Add "Patch by Roy Williams."

"... when compiling a  bytes instance with misplaced inline modifier." Or something like.

And usually new entries are added at the begin of a section, so they are listed in the reversed chronological order.

**Review by serhiy-storchaka — CHANGES_REQUESTED** [review]

**Comment by rowillia** [comment]
Thanks again @serhiy-storchaka for taking a look.  I promise usually I spell my own name correctly 😆 !

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by serhiy-storchaka** [comment]
Thank you for your contribution @rowillia. Do you mind to backport this fix to 3.6?

**Comment by rowillia** [comment]
Sounds good to me @serhiy-storchaka, thanks for your help landing this.

**Pull request merged**